source/common/tcp_proxy/upstream.cc

```diff
@@ -140,13 +140,10 @@ void HttpUpstream::encodeData(Buffer::Instance& data, bool end_stream) {
   if (!request_encoder_) {
     return;
   }
+  auto codec = type_;
```

Comment:
what's the reason we can't just use type_ below inline? is it expected to change?

Reply:
In H1 codec, if `end_stream` is received it [triggers downstream side resets](https://github.com/envoyproxy/envoy/blob/main/source/common/http/http1/codec_impl.cc#L258) which was reseting current HttpUpstream instance pointer as a result member `type_` was going out of scope and could not be used in the following lines to skip `doneWriting` in H1 case.

Now I am handling it by not reseting `upstream_` immediately rather adding it to defer delete list.